async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -42,17 +42,18 @@ object AsyncHttpClient {
     */
   def apply(config: AsyncHttpClientConfig = defaultConfig,
             bufferSize: Int = 8,
-            executorService: ExecutorService = DefaultExecutor.newClientDefaultExecutorService("async-http-client-response")): Client = {
+            customExecutor: Option[ExecutorService] = None): Client = {
     val client = new DefaultAsyncHttpClient(config)
-    val close = executorService match {
-      case es: DefaultExecutorService =>
+    val executorService = customExecutor.getOrElse(DefaultExecutor.newClientDefaultExecutorService("async-http-client-response"))
+    val close =
+      if (customExecutor.isDefined)
```

Comment:
Doing this made me realize I have no idea how to format a multi-line if/else in Scala. I guess I hardly ever use if/else!